test/ok/error_test.exs

```diff
@@ -0,0 +1,35 @@
+defmodule OK.ErrorTest do
+  use ExUnit.Case
+
+  def foo(x) do
+    {:ok, x}
+  end
+
+  def bar(y) do
+    {:bad, y}
+  end
+
+  require OK
+
+  test "helpful error for binding failure" do
+    message = """
+    Binding to variable failed, '{:bad, 6}' is not a result tuple.
+
+        Code
+          b <- bar(a)
+
+        Expected signature
+          bar(a) :: {:ok, b} | {:error, reason}
+
+        Actual values
+          bar(a) :: {:bad, 6}
+    """
+    assert_raise OK.BindError, message, fn() ->
+      OK.with do
+        a <- foo(6)
+        b <- bar(a)
+        _ <- {:ok, b}
```

Comment:
Overall, I think it's great. I love how it reports the actual line where the result tuple doesn't match. But I did come across two things: 

1. This is causing a compiler warning for me: 
```
warning: this clause cannot match because of different types/sizes
  test/ok/error_test.exs:31

warning: this clause cannot match because a previous clause at line 31 always matches
  test/ok/error_test.exs:31
```
2. In exploring the above, I tried the following, which works fine and produces no compiler warnings:
```
  test "helpful error for binding failure" do
    message = """
    Binding to variable failed, '{:bad, 6}' is not a result tuple.

        Code
          _ <- bar(a)

        Expected signature
          bar(a) :: {:ok, _} | {:error, reason}

        Actual values
          bar(a) :: {:bad, 6}
    """
    assert_raise OK.BindError, message, fn() ->
      OK.with do
        a <- foo(6)
        _ <- bar(a)
      end
    end
  end
end
```

But it occurred to me to check this without the seemingly unnecessary `_ <-` assignment:

```
OK.with do
  a <- foo(6)
  bar(a)
end
```

This produces a `CaseClauseError` instead of an `OK.BindError` which doesn't seem quite right to me.